File: ForumAPI/ForumAPI/DTOs/GreenitorDTOs/RegisterUserDTO.cs
namespace ForumAPI.DTOs.GreenitorDTOs
{
    /// <summary>
    /// Data transfer object for user registration.
    /// </summary>
    public class RegisterUserDTO
    {
        /// <summary>
        /// Gets or sets the username for registering the user.
        /// </summary>
        public string Username { get; set; }

        /// <summary>
        /// Gets or sets the email address for registering the user.
        /// </summary>
        public string Email { get; set; }

        /// <summary>
        /// Gets or sets the password for registering the user.
        /// </summary>
        public string Password { get; set; }

        /// <summary>
        /// Gets or sets the profile image for the user during registration.
        /// This property is optional and may not be provided.
        /// </summary>
        public IFormFile? Image { get; set; }
    }
}

File: ClientAPI/src/main/java/pt/upskill/clientapi/DTOs/RegisterUserDTO.java
package pt.upskill.clientapi.DTOs;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Data Transfer Object for registering a new user.
 * This class is used to hold the necessary information for registering a new user.
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class RegisterUserDTO {

    /**
     * The username of the user to be registered.
     * This will be used as the unique identifier for the user.
     */
    private String username;

    /**
     * The email of the user to be registered.
     * This will be used for communication and possibly authentication.
     */
    private String email;

    /**
     * The password of the user to be registered.
     * This is used for authenticating the user during login.
     */
    private String password;

    /**
     * The image URL or path associated with the user to be registered.
     * This can be a link to the user's avatar or profile image.
     */
    private String image;

}